storage/rocksdb/ha_rocksdb.cc

```diff
@@ -5479,12 +5525,12 @@ class Rdb_transaction_impl : public Rdb_transaction {
   }
 
   virtual ~Rdb_transaction_impl() override {
+    rollback();
```

Comment:
What will happen when we finish a transaction? My understanding is this transaction will be first reinsert without snapshot to the end of the list in on_commit(). Then when the transaction is deconstructed, it will be moved to the end again in the rollback(). Lastly, it will be removed from the list via erase(). Did I miss something, or we do have duplicated/redundant operations?

Reply:
The destructor is only executed on a session disconnect, where it may be called with or without going through commit/rollback first. While a session is alive, the transaction object does not go through destruction but is reused after completing rollback/commit.